### DISCUSSION TITLE
[Bug] username/email field case sensitive

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/full-stack-fastapi-template#2118] [Questions]
Out-of-the-box the username/email field is case sensitive. A single email could create many users by changing the capitalization.

i.e. Test@test.com + test@test.com

I think the best way to handle this is to save the case-sensitive user/email when creating the user, and then when logging in or creating a new user, if there is already a case-insensitive matching user, it will log in to that one or, if creating a new account, reject the request to create a new user because the user already exists.

**Comment by tiangolo** [comment]
Emails, by the standards, are not case insensitive, even though several providers treat them as such. So I'll pass on this request for now.
